ui/app/components/app/confirm-page-container/confirm-detail-row/tests/confirm-detail-row.component.test.js

```diff
@@ -1,64 +1,73 @@
 import assert from 'assert'
 import React from 'react'
-import { shallow } from 'enzyme'
 import sinon from 'sinon'
+import configureMockStore from 'redux-mock-store'
+import { fireEvent } from '@testing-library/react'
+import render from '../../../../../../../test/lib/render-helpers'
 import ConfirmDetailRow from '../confirm-detail-row.component'
 
-const propsMethodSpies = {
-  onHeaderClick: sinon.spy(),
-}
-
 describe('Confirm Detail Row Component', function () {
-  describe('render', function () {
-    let wrapper
+  let utils
 
-    beforeEach(function () {
-      wrapper = shallow(
-        <ConfirmDetailRow
-          errorType="mockErrorType"
-          label="mockLabel"
-          showError={false}
-          primaryText="mockFiatText"
-          secondaryText="mockEthText"
-          primaryValueTextColor="mockColor"
-          onHeaderClick={propsMethodSpies.onHeaderClick}
-          headerText="mockHeaderText"
-          headerTextClassName="mockHeaderClass"
-        />,
-      )
-    })
+  const props = {
+    errorType: 'mockErrorType',
+    label: 'mockLabel',
+    showError: false,
+    primaryText: 'mockFiatText',
+    secondaryText: 'mockEthText',
+    primaryValueTextColor: 'mockColor',
+    onHeaderClick: sinon.spy(),
+    headerText: 'mockHeaderText',
+    headerTextClassName: 'mockHeaderClass',
+  }
 
-    it('should render a div with a confirm-detail-row class', function () {
-      assert.equal(wrapper.find('div.confirm-detail-row').length, 1)
-    })
+  const mockState = {
+    metamask: {
+      conversionRate: '100',
+      currentCurrency: 'used',
+      provider: {
+        type: 'mainnet',
+      },
+      preferences: {
+        useNativeCurrencyAsPrimaryCurrency: true,
+      },
+    },
+  }
 
-    it('should render the label as a child of the confirm-detail-row__label', function () {
-      assert.equal(wrapper.find('.confirm-detail-row > .confirm-detail-row__label').childAt(0).text(), 'mockLabel')
-    })
+  const store = configureMockStore()(mockState)
 
-    it('should render the headerText as a child of the confirm-detail-row__header-text', function () {
-      assert.equal(wrapper.find('.confirm-detail-row__details > .confirm-detail-row__header-text').childAt(0).text(), 'mockHeaderText')
-    })
+  beforeEach(function () {
+    utils = render(<ConfirmDetailRow {...props} />, store)
+  })
 
-    it('should render the primaryText as a child of the confirm-detail-row__primary', function () {
-      assert.equal(wrapper.find('.confirm-detail-row__details > .confirm-detail-row__primary').childAt(0).text(), 'mockFiatText')
-    })
+  after(function () {
```

Comment:
Shouldn't this be `afterEach`? That is the pattern I am hoping we can move toward, so that we can eventually add a top-level `afterEach` hook for the entire test suite, [as suggested by the Sinon docs](https://sinonjs.org/releases/v9.2.0/general-setup/).